### PR TITLE
Add option to run custom architecture (container platform)

### LINKED ISSUE #493: enable specifying docker --platform argument
Context: For folks on M1 macs that was to run actions as amd64.

### PR DESCRIPTION
**Pull request by catthehacker**
Closes #493

### CODE REVIEW

**Comment by catthehacker** [comment]
Ok, I have currently no idea where it fails...

**Comment by catthehacker** [comment]
Current test will fail as it's dependent on #524

File: pkg/runner/run_context.go [hunk 1]
```diff
@@ -89,6 +89,14 @@ func (rc *RunContext) startJobContainer() common.Executor {
 			binds = append(binds, fmt.Sprintf("%s:%s%s", rc.Config.Workdir, "/github/workspace", bindModifiers))
 		}
 
+		if rc.Config.ContainerPlatform == "" {
+			if runtime.GOOS == "darwin" {
+				rc.Config.ContainerPlatform = fmt.Sprintf("%s/%s", "darwin", runtime.GOARCH)
```
Comment (catthehacker) [inline pkg/runner/run_context.go:94]: `darwin/*` platform doesn't exists 🤦 will fix that later

**Comment by catthehacker** [comment]
@cplee Let me know if there is anything wrong.
I've also renamed the parameter lately to `--container-architecture` since `platform` could be confused with `-P, --platform stringArray   custom image to use per platform`.
Whenever it's mentioned in `act` context it's `containerArchitecture` and when it touches Docker API it's `Platform`.

**Review by cplee — APPROVED** [review]

File: pkg/runner/step_context.go [hunk 2]
```diff
@@ -354,10 +359,35 @@ func (sc *StepContext) runAction(actionDir string, actionPath string) common.Exe
...
+					Platform:   rc.Config.ContainerArchitecture,
 				})
+				exists, err = container.ImageExistsLocally(ctx, image, rc.Config.ContainerArchitecture)
+				if err != nil {
+					return err
+				}
+
+				if !exists {
+					return err
+				}
```
Comment (ZauberNerd) [inline pkg/runner/step_context.go:390]: The code always exits here with err `nil` and the code below won't get executed, which causes local docker actions to be skipped.
This is caused by `container.NewDockerBuildExecutor` returning a function which is not executed here, but in line 483.
/cc @catthehacker @cplee

**Pull request merged**